### PR TITLE
[css-values-5] Introduce `sibling-index()` and `sibling-count()` (#4559)

### PR DESCRIPTION
**Pull request by bramus**
A first, and possible naive, attempt at speccing `sibling-index()` and `sibling-count()`. Very much open to feedback on how to do this properly.

### CODE REVIEW

File: css-values-5/Overview.bs [hunk 1]
```diff
@@ -945,8 +945,47 @@ Generating/Caching Random Values: the <<random-caching-options>> value</h4>
...
+
+<h3 id=random>
+Getting the number of siblings an element is part of: the ''sibling-count()'' function</h3>
+
 
+	The <dfn>sibling-count()</dfn> function resolves to the total number of
+	<a>inclusive siblings</a> an element is part of. That is, it resolves to 
+	the total number of direct children the element’s parent element contains.
 
+Issue: Describe when it actually resolves. Computed Value? Used Value?
```
Comment (andruud) [inline css-values-5/Overview.bs:988]: Definitely resolves computed-value-time, no reason to wait any longer.

File: css-values-5/Overview.bs [hunk 2]
```diff
@@ -945,8 +945,47 @@ Generating/Caching Random Values: the <<random-caching-options>> value</h4>
...
+	allow authors to take these factors into account in ''calc()'' expressions.
+
+	With these, authors can create things like staggered animation delays
+	or easily position elements on a circle.
+
+<h3 id=random>
+Getting the position of an element within its siblings: the ''sibling-index()'' function</h3>
+
+	The <dfn>sibling-index()</dfn> function resolves to the position
+	of an element within its group of <a>inclusive siblings</a>.
```
Comment (andruud) [inline css-values-5/Overview.bs:975]: Might want to clarify that it's the inclusive _flat tree_ siblings, which is what we want (I assume).

File: css-values-5/Overview.bs (hunk 1 above)
Comment (bramus) [reply css-values-5/Overview.bs]: Added in af5ce47

File: css-values-5/Overview.bs (hunk 2 above)
Comment (bramus) [reply css-values-5/Overview.bs]: Clarified in 30de15b, but not sure if my wording is 100% correct.

Comment (lilles) [reply css-values-5/Overview.bs]: We don't use flat tree order for :nth-child(). I think we want :nth-*() and sibling-index() to be consistent?

Comment (andruud) [reply css-values-5/Overview.bs]: That's true.

File: css-values-5/Overview.bs [hunk 3]
```diff
@@ -945,8 +945,51 @@ Generating/Caching Random Values: the <<random-caching-options>> value</h4>
...
+
+<h2 id=element-counting>
+Counting Elements</h2>
+
+	Authors often want to apply element specific styles
+	based on the position of the element within the tree.
+
+	The ''sibling-index()'' and ''sibling-count()'' functions	
+	(the <dfn export for=CSS>element counting functions</dfn>)
+	allow authors to take these factors into account in ''calc()'' expressions.
```
[the remark this answers is not in the document]
Comment (tabatkins) [reply css-values-5/Overview.bs:966]: (Just a light rewrite to avoid implying that these are only valid in calc().)

**Review by tabatkins — CHANGES_REQUESTED** [review]
A couple of nits. Also, I'd add a few small examples to show them off.

**Comment by Loirooriol** [comment]
With 762eacee436fdcb1b11e7e482e6cdb3fee161317, I guess this can be closed.

**Comment by bramus** [comment]
Yes, thanks for closing this one @Loirooriol. I was preoccupied with Other Important Things™ + some time OOO, so never got round to reviewing the suggested changes in this thread.

**Pull request closed without merging**